### DISCUSSION TITLE
[Question] Stop action noises

### DISCUSSION THREAD

**Discussion by vpoguru** [GrandOrgue/grandorgue#841] [Help]
I implemented stop action noises as follows:

    [Stop801]
    Name=Subbass 16' (Pedal) Action Noise Engage Stop
    NumberOfLogicalPipes=1
    NumberOfAccessiblePipes=1
    FirstAccessiblePipeLogicalPipeNumber=001
    FirstAccessiblePipeLogicalKeyNumber=001
    WindchestGroup=004
    Percussive=Y
    Function=And
    SwitchCount=2
    Switch001=001 ; Subbass 16'
    Switch002=025 ; Action Noise
    Pipe001=./actionNoises/stops_engage/036-C.wav

    [Stop901]
    Name=Subbass 16' (Pedal) Action Noise Disengage Stop
    NumberOfLogicalPipes=1
    NumberOfAccessiblePipes=1
    FirstAccessiblePipeLogicalPipeNumber=001
    FirstAccessiblePipeLogicalKeyNumber=001
    WindchestGroup=004
    Percussive=Y
    Function=Nand
    SwitchCount=2
    Switch001=001 ; Subbass 16'
    Switch002=025 ; Action Noise
    Pipe001=./actionNoises/stops_disengage/036-C.wav

_Switch001_ is the switch, which is connected to a virtual control that activates the stop. _Switch025_ is a hidden switch with _DefaultToEngaged=Y_.

This implementation works. However, whenever I startup the sample set a lot noise is heard, because ALL stops are disengaged and probably trigger the disengage noise.

Is there a smarter way to implement the stop action noises? Or is it somehow possible to suppress the startup noise?

**Comment by larspalo** [comment]
The way you've done it with the logical switches you'll always have a noise happening as one is triggered when the Switch001 is on and the other when it's off.

I usually have effect noises like this with a loop and release instead (implies Percussive=N) which makes for much simpler logic. If your setup indeed has different sound files for on and off you can follow my suggestion in #831 and achieve the same result.

Actually, when I think about it, I wrote a program called "noisepipes" a long time ago that can be found at https://github.com/e9925248/grandorgue-odf/tree/master/noisepipes or https://github.com/larspalo/grandorgue-odf/tree/master/noisepipes that writes out the necessary lines to add for getting a separate attack with looped last sample and a separate release combined. Might be something to check out... I used it to create action noises for a SP sampleset that had separate attacks and releases!
